packages/e2e-tests/specs/editor/various/datepicker.test.js

```diff
@@ -1,88 +1,137 @@
 /**
  * WordPress dependencies
  */
-import { createNewPost } from '@wordpress/e2e-test-utils';
+import { createNewPost, changeSiteTimezone } from '@wordpress/e2e-test-utils';
+
+async function getInputValue( selector ) {
+	return page.$eval( selector, ( element ) => element.value );
+}
+
+async function getSelectedOptionLabel( selector ) {
+	return page.$eval(
+		selector,
+		( element ) => element.options[ element.selectedIndex ].text
+	);
+}
+
+async function getDatePickerValues() {
+	const year = await getInputValue( '[aria-label="Year"]' );
+	const month = await getInputValue( '[aria-label="Month"]' );
+	const monthLabel = await getSelectedOptionLabel( '[aria-label="Month"]' );
+	const day = await getInputValue( '[aria-label="Day"]' );
+	const hours = await getInputValue( '[aria-label="Hours"]' );
+	const minutes = await getInputValue( '[aria-label="Minutes"]' );
+	const amOrPm = await page.$eval(
+		'.components-datetime__time-field-am-pm .is-primary',
+		( element ) => element.innerText.toLowerCase()
+	);
+
+	return { year, month, monthLabel, day, hours, minutes, amOrPm };
+}
+
+function trimLeadingZero( str ) {
+	return str[ 0 ] === '0' ? str.slice( 1 ) : str;
+}
+
+function formatDatePickerValues( {
+	year,
+	monthLabel,
+	day,
+	hours,
+	minutes,
+	amOrPm,
+} ) {
+	const dayTrimmed = trimLeadingZero( day );
+	const hoursTrimmed = trimLeadingZero( hours );
+	return `${ monthLabel } ${ dayTrimmed }, ${ year } ${ hoursTrimmed }:${ minutes } ${ amOrPm }`;
+}
+
+async function getPublishingDate() {
+	return page.$eval(
+		'.edit-post-post-schedule__toggle',
+		( dateLabel ) => dateLabel.textContent
+	);
+}
 
 describe( 'Datepicker', () => {
-	beforeEach( async () => {
-		await createNewPost();
-	} );
+	[ 'UTC-10', 'UTC', 'UTC+10' ].forEach( ( timezone ) => {
+		describe( timezone, () => {
+			let oldTimezone;
+			beforeEach( async () => {
+				oldTimezone = await changeSiteTimezone( timezone );
+				await createNewPost();
+			} );
+			afterEach( async () => {
+				await changeSiteTimezone( oldTimezone );
+			} );
```

Comment:
Changing this to 

```suggestion
			beforeEach( async () => {
				await changeSiteTimezone( timezone );
				await createNewPost();
			} );
```

seems to have the same effect—perhaps I'm missing something?

Reply:
It has the same effect, except when one of the tests fails. In that case, the timezone is not reset to the original timezone. Also, if we don't reset the timezone then the rest of the tests will use the last set timezone.

It's not mandatory to reset the timezone though. Since tests should pass in whatever timezone we are currently in.

What do you think, should we timezone resetting? 
**Pros**: Less code 
**Cons**: Tests are less consistent, sometimes they might run in a different timezone

Reply:
I don't think the benefit of less code outweighs consistency, so I'd keep it as it is. Thanks for clarifying :)